lib/App/GHPT/WorkSubmitter.pm

```diff
@@ -105,15 +131,56 @@ has _project_ids => (
 );
 
 sub _build_project_ids ($self) {
+    my $projects = $self->_pt_api->projects;
+
     my $want = $self->project;
     if ($want) {
-        for my $project ( $self->_pt_api->projects->@* ) {
+        for my $project ( $projects->@* ) {
             return [ $project->id ] if $project->name =~ /\Q$want/i;
         }
-        die 'Could not find a project id for project named ' . $self->project;
+        die 'Could not find a project id for project named ' . $want . "\n";
     }
 
-    return [ map { $_->id } $self->_pt_api->projects->@* ];
+    return [ map { $_->id } $projects->@* ];
+}
+
+sub _find_project ($self) {
+    my $projects = $self->_pt_api->projects;
+
+    my $want = $self->project;
+    if ($want) {
+        for my $project ( $projects->@* ) {
+            return $project if $project->name =~ /\Q$want/i;
+        }
+        die 'Could not find a project id for project named ' . $want . "\n";
+    }
+
+    my %project_by_name
+        = map { $_->name . ' (ID ' . $_->id . ')' => $_ } $projects->@*;
+    my $name = $self->_choose( [ sort keys %project_by_name ] );
+    return $project_by_name{$name};
+}
+
+sub _find_requester ( $self, $project ) {
+    my $memberships = $project->memberships;
+
+    my $want = $self->requester;
+    if ($want) {
+        for my $membership ( $memberships->@* ) {
+            my $got_name  = fc( $membership->person->name );
+            my $want_name = fc($want);
+            return $membership->person
```

Comment:
This matching isn't a good general case solution.  It won't be possible to create something reliably for user "mark" if user "markfowler" also exists.